### PR TITLE
add op_check_not_exists

### PR DESCRIPTION
**Pull request by lysu**
just like `op_insert` is "insert should not exists", we need another "delete should not exists".

https://github.com/pingcap/tidb/pull/14968

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/kvproto/565)

### CODE REVIEW

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -163,6 +163,7 @@ enum Op {
     // insert operation has a constraint that key should not exist before.
     Insert = 4;
     PessimisticLock = 5;
+    CheckExists = 6;
```
Comment (coocood) [inline proto/kvrpcpb.proto]: Have you considered `CheckNotExists`?

Comment (lysu) [reply proto/kvrpcpb.proto]: yes, it should be CheckNotExists, my fault- -

**Comment by coocood** [comment]
LGTM

**Comment by lysu** [comment]
hi @coocood, I found a but last night, we also need write key, nil when key not exists.

to avoid txnNotFoundError when `begin; insert record1;delete record1; commit`(we need prewrite something to help commit can works, maybe better way is check prewrite type and avoid commit phase, but that's more complex),

**Comment by lysu** [comment]
@jackysp  PTAL if free, kvproto need two approve, then we can continue work on tidb/tikv pr

**Comment by lysu** [comment]
now tidb can ensure never pass checkNotExists key into commit phase

**Comment by coocood** [comment]
LGTM

**Comment by jackysp** [comment]
@coocood please approve the pr.

File: proto/kvrpcpb.proto [hunk 2]
```diff
@@ -163,6 +163,7 @@ enum Op {
     // insert operation has a constraint that key should not exist before.
     Insert = 4;
     PessimisticLock = 5;
+    CheckNotExists = 6;
```
Comment (youjiali1995) [inline proto/kvrpcpb.proto]: Why use the third person singular?😂

**Pull request merged**